src/backend/seq/pup/yaksuri_seq_pup.c

```diff
@@ -29,15 +29,17 @@ int yaksuri_seq_ipack(const void *inbuf, void *outbuf, uintptr_t count, yaksi_ty
 {
     int rc = YAKSA_SUCCESS;
     yaksuri_seqi_type_s *seq_type = (yaksuri_seqi_type_s *) type->backend.seq.priv;
+    int memcpy_type = YAKSU_MEMCPY__SYSTEM;
 
     uintptr_t iov_pack_threshold = YAKSURI_SEQI_INFO__DEFAULT_IOV_PUP_THRESHOLD;
     if (info) {
         yaksuri_seqi_info_s *seq_info = (yaksuri_seqi_info_s *) info->backend.seq.priv;
         iov_pack_threshold = seq_info->iov_pack_threshold;
+        if (seq_info->stream_cpy) memcpy_type = YAKSU_MEMCPY__STREAM;
     }
 
     if (op == YAKSA_OP__REPLACE && type->is_contig) {
-        memcpy(outbuf, (const char *) inbuf + type->true_lb, type->size * count);
+        yaksu_memcpy(outbuf, (const char *)inbuf + type->true_lb, type->size * count, memcpy_type);
```

Comment:
Isn't the contig case bypassed in MPICH already with `MPIR_Memcpy`?

Reply:
Yes. Though it could presumably be done a different way outside of MPICH. This code would work for either case.

Reply:
I am really reluctant to have duplicated code in two places, so if it is unnecessary for now, I would delay it.

Reply:
Do you mean this entire PR or just this line?

Reply:
I think the entire PR is unnecessary for now.

Reply:
It's still useful if you want to accelerate non-contiguous copies. The code in MPICH by itself doesn't handle that situation.

Reply:
Only the low-density datatype is handled in this PR anyway. MPICH can handle that as well if it wants to, and it will probably be simpler than maintaining a duplicate code in yaksa and the extra hint.

Reply:
I'm not very familiar with the Yaksa code so I probably missed something here. Where else should I look to handle the remaining cases?

Reply:
They are in the python-generated "pup" functions, in which they are assignments within loops.

Reply:
I see. That's why I didn't find it then. I'll see if I can figure out what needs to change there.